src/coprocessor/metrics.rs

```diff
@@ -98,3 +98,48 @@ lazy_static! {
             20.0, 24.0, 28.0, 32.0, 48.0, 64.0, 96.0, 128.0, 192.0, 256.0]
         ).unwrap();
 }
+
+/// `ScanCounter` is for recording range query and point query.
+pub struct ScanCounter {
+    range: usize,
+    point: usize,
+}
+
+impl ScanCounter {
+    pub fn new() -> Self {
+        ScanCounter {
+            range: 0,
+            point: 0,
+        }
+    }
+
+    #[inline]
+    pub fn inc_range(&mut self) {
+        self.range += 1;
+    }
+
+    #[inline]
+    pub fn inc_point(&mut self) {
+        self.point += 1;
+    }
+
+    pub fn merge(&mut self, other: &mut ScanCounter) {
```

Comment:
I perfer to name it like  `collect..` since you've clear `other`

Reply:
same strange here, seem this is not for merge.